Src/DependencyCollector/Shared/Implementation/HttpDiagnosticSourceListener.cs

```diff
@@ -2,31 +2,29 @@
 namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
 {
     using System;
-    using System.Collections.Concurrent;
     using System.Collections.Generic;
-    using System.Diagnostics;
-    using System.Globalization;
-    using System.Linq;
     using System.Net;
-    using Microsoft.ApplicationInsights.Common;
-    using Microsoft.ApplicationInsights.DataContracts;
-    using Microsoft.ApplicationInsights.Extensibility;
-    using Microsoft.ApplicationInsights.Extensibility.Implementation;
-    using Microsoft.ApplicationInsights.Web.Implementation;
+    using System.Reflection;
 
     /// <summary>
     /// Diagnostic listener implementation that listens for Http DiagnosticSource to see all outgoing HTTP dependency requests.
     /// </summary>
     internal class HttpDiagnosticSourceListener : IObserver<KeyValuePair<string, object>>, IDisposable
     {
         private readonly FrameworkHttpProcessing httpProcessingFramework;
-        private HttpDiagnosticSourceSubscriber subscribeHelper;
+        private readonly HttpDiagnosticSourceSubscriber subscribeHelper;
+        private readonly PropertyFetcher requestFetcherRequestEvent;
+        private readonly PropertyFetcher requestFetcherResponseEvent;
+        private readonly PropertyFetcher responseFetcher;
         private bool disposed = false;
 
         internal HttpDiagnosticSourceListener(FrameworkHttpProcessing httpProcessing)
         {
             this.httpProcessingFramework = httpProcessing;
             this.subscribeHelper = new HttpDiagnosticSourceSubscriber(this);
+            this.requestFetcherRequestEvent = new PropertyFetcher("Request");
+            this.requestFetcherResponseEvent = new PropertyFetcher("Request");
```

Comment:
```
 case "System.Net.Http.Request":
                    {
                        var request = (HttpWebRequest)this.requestFetcherRequestEvent.Fetch(value.Value);
                        this.httpProcessingFramework.OnRequestSend(request);
                        break;
                    }

                    case "System.Net.Http.Response":
                    {
                        var request = (HttpWebRequest)this.requestFetcherResponseEvent.Fetch(value.Value);
                        var response = (HttpWebResponse)this.responseFetcher.Fetch(value.Value);
                        this.httpProcessingFramework.OnResponseReceive(request, response);
                        break;
                    }
```
* requestFetcher**Request**Event reads `Request` property from the `System.Net.Http.Request` event payload
* requestFetcher**Response**Event reads `Request` property from `System.Net.Http.Response` event payload
* responseFetcher reads `Response` property from the `System.Net.Http.Response` event payload.

If you want to propose better names for the fetchers, I'll be happy to do so :)